src/components/Slider/Slider.tsx

```diff
@@ -8,53 +8,58 @@ import { ArrowDirection } from './constants/constants';
 import 'swiper/swiper-bundle.css';
 import './Slider.scss';
 import { useSelectorTyped } from './redux/store';
-import { useDispatch } from 'react-redux';
 import { Endpoints } from '../../shared/api/constants/endpoints';
-import { Slides } from './types/SliderTypes';
+import { SliderAnyActions, Slides } from './types/SliderTypes';
 import { getSlides } from './services/getSlides';
+import { useDispatch } from 'react-redux';
+import { Dispatch } from 'redux';
 
 SwiperCore.use([Navigation, Autoplay]);
 
 export const Slider = () => {
-  const { slides, error, isLoading } = useSelectorTyped(state => state);
-  const dispatch = useDispatch();
+  const { slides, error, isLoading } = useSelectorTyped(state => state.slider);
+  const dispatch = useDispatch<Dispatch<SliderAnyActions>>();
 
   useEffect(() => {
-    dispatch(
-      getSlides(
-        Endpoints.GO_FOR_LAUNCHES_ENDPOINT,
-        Endpoints.PREVIOUS_LAUNCHES_ENDPOINT,
-      ),
-    );
+    getSlides(
+      Endpoints.GO_FOR_LAUNCHES_ENDPOINT,
+      Endpoints.PREVIOUS_LAUNCHES_ENDPOINT,
+    )(dispatch);
   }, []);
```

Comment:
add dispatch to useEffect dependencies, it allows you to render only at dispatch action